data/data_interface.py

```diff
@@ -0,0 +1,184 @@
+import pandas as pd
+from typing import Sequence, Any, Tuple
+import os
+import pathlib
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler, MinMaxScaler
+from synthetic_data import create_synthetic_data
+
+class DataInterface():
+    def __init__(self, data: pd.DataFrame, file_path: str, continuous_features: Sequence[str],
+                ordinal_features: Sequence[str], categorical_features: Sequence[str], immutable_features: Sequence[str],
+                target_feature: str, target_mapping: Any = None, scaling_method: str = "MinMax", encoding_method: str = "OneHot",
+                pos_label: int = 1):
+        """
+        Creates a data interface with the specified, continuous features, ordinal features, and categorical features.
+
+        Features can be normalized, standardized, or encoded as desired. Additionally data can be split into train and test data.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            A dataframe of the data to load. If None, we will attempt to load using file_path
+        file_path : str
+            The path to load your data from. Not required with data is not None.
+        continuous_features : Sequence[str]
+            A sequence of the continous features.
+        ordinal_features : Sequence[str]
+            A sequence of the ordinal features.
+        categorical_features : Sequence[str]
+            A sequence of the categorical/nominal features.
+        immutable_features : Sequence[str]
+            A sequence of the immutable features.
+        target_feature : str
+            The column name of the target feature
+        target_mapping : Any
+            A mapping of the target features to some values
+        scaling_method : str
+            String indicating how you with to scale your data. Options are MaxMin normalization and standardization 
+        encoding_method : str
+            Encoding method for categorical features. Only one-hot for now.
+        """
+        self._continuous_features = continuous_features
+        self._ordinal_features =  ordinal_features
+        self._categorical_features = categorical_features
+        self._immutable_features = immutable_features
+        self._target_feature = target_feature
+        self._target_mapping = target_mapping
+        self._scaling_method = scaling_method
+        self._encoding_method = encoding_method
+        if (file_path is None or file_path == "") and isinstance(data, pd.DataFrame):
+            df = data
+        elif file_path is not None and file_path != "":
+            df = pd.read_csv(file_path)
+        else:
+            raise Exception("No data and file path provided")
+        X_features = df.columns[df.columns != target_feature]
+        self._df_y = df[self._target_feature]
+        self._df_X = df[X_features]
+        self._df_y_init = self._df_y.copy()

```

Comment:
What was the purpose of the two DataFrame copies? Do we need this right now? If not, let's remove this and we can always add this later.